Projects/Addons/Tests/ViewModelTests/AddonsViewModelTests.swift

```diff
@@ -0,0 +1,225 @@
+@preconcurrency import XCTest
+import hCore
+
+@testable import Addons
+
+@MainActor
+final class AddonsViewModelTests: XCTestCase {
+    weak var sut: MockAddonsService?
+    weak var vm: ChangeAddonViewModel?
+
+    override func setUp() async throws {
+        try await super.setUp()
+        sut = nil
+    }
+
+    override func tearDown() async throws {
+        Dependencies.shared.remove(for: AddonsClient.self)
+        try await Task.sleep(nanoseconds: 20_000_000)
+        XCTAssertNil(sut)
+        XCTAssertNil(vm)
+    }
+
+    func testFetchAddonSuccess() async throws {
+        let selectedSubOption = AddonSubOptionModel(
+            id: "subOptionId",
+            title: "sub option title",
+            subtitle: "sub option subTitle",
+            price: .init(amount: "79", currency: "SEK")
+        )
+
+        let addonModel: AddonModel = .init(
+            id: "addonId",
+            title: "title",
+            subTitle: "subTitle",
+            tag: "tag",
+            informationText: "information text",
+            options: [
+                .init(
+                    id: "optionId",
+                    title: "option title",
+                    subtitle: "option subTitle",
+                    price: .init(amount: "49", currency: "SEK"),
+                    subOptions: [
+                        selectedSubOption
+                    ]
+                )
+            ]
+        )
+
+        let mockService = MockData.createMockAddonsService(fetchAddon: {
+            addonModel
+        })
+
+        self.sut = mockService
+
+        let model = ChangeAddonViewModel(contractId: "contractId")
+        model.fetchAddonsViewState = .loading
```

Comment:
Why do we set this property to .loading? It looks like it has that value by default